payment-handler/can-make-payment-event.https.html

```diff
@@ -0,0 +1,98 @@
+    supportedMethods: methodName,
+  }], {
+    total: {
+      label: 'Total',
+      amount: {
+        currency: 'USD',
+        value: '0',
+      },
+    },
+  });
+  return request;
+}
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-false';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = false;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_false(paymentRequestCanMakePaymentResult, 'canMakePayment() should return false.');
+  await promise_rejects(t, 'NotSupportedError', request.show());
+}, 'If CanMakePaymentEvent.respondWith(false) is called, then the payment method is not supported.');
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-promise-false';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = false;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_false(paymentRequestCanMakePaymentResult, 'canMakePayment() should return false.');
+  await promise_rejects(t, 'NotSupportedError', request.show());
+}, 'If CanMakePaymentEvent.respondWith(Promise.resolve(false)) is called, then the payment method is not supported.');
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-true';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = true;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_true(paymentRequestCanMakePaymentResult, 'canMakePayment() should return true.');
+  const acceptPromise = request.show();
+  await request.abort();
+  await promise_rejects(t, 'AbortError', acceptPromise);
+}, 'If CanMakePaymentEvent.respondWith(true) is called, then the payment method is supported.');
+
+promise_test(async t => {
+  const methodName = window.location.origin + '/canMakePayment-promise-true';
+  const request = await registerAppAndBuildPaymentRequest(methodName);
+  let paymentRequestCanMakePaymentResult = true;
+  try {
+    paymentRequestCanMakePaymentResult = await request.canMakePayment();
+  } catch (err) {
+    assert_equals(err.name, 'NotAllowedError', 'If it throws, then it must be NotAllowedError');
+  }
+  assert_true(paymentRequestCanMakePaymentResult, 'canMakePayment() should return true.');
+  const acceptPromise = request.show();
+  await request.abort();
+  await promise_rejects(t, 'AbortError', acceptPromise);
```

Comment:
Nit: I wonder if we should bother having this as part of the test? We tested this fairly well in the Payment Request tests.

Reply:
To be clear, I mean including this line: 

```
await promise_rejects(t, 'AbortError', acceptPromise);
```

Applied similarly to other tests above.

Reply:
If I don't have a `promise_rejects` here, then the test shows an error message:

> Uncaught (in promise) AbortError: The website has aborted the payment.

I would like to avoid this error and `promise_rejects` seems like one good way to achieve this.